benchmarks/src/main/java/io/grpc/benchmarks/qps/ClientConfiguration.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2015, Google Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *    * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *    * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ *    * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.grpc.benchmarks.qps;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.lang.Math.max;
+
+import grpc.testing.Qpstest.PayloadType;
+import grpc.testing.Qpstest.RpcType;
+
+class ClientConfiguration {
+
+  boolean okhttp;
+  boolean tls;
+  boolean testca;
+  boolean directExecutor;
+  boolean dumpHistogram;
+  int port;
+  int channels = 4;
+  int outstandingRpcsPerChannel = 10;
+  int serverPayload;
+  int clientPayload;
+  // seconds
+  int duration = 60;
+  // seconds
+  int warmupDuration = 10;
+  String host = "127.0.0.1";
+  String histogramFile = "latencies.hgrm";
+  RpcType rpcType = RpcType.UNARY;
+  PayloadType payloadType = PayloadType.COMPRESSABLE;
+
+  ClientConfiguration() {
+  }
+
+  ClientConfiguration (String[] args) {
+    parseArgs(checkNotNull(args, "args"));
+  }
+
+  private void parseArgs(String[] args) {
```

Comment:
Instead of having a constructor that takes the arguments, could you just make parseArgs a static factory method:

```java
// Throws if an error occurred parsing the arguments.
public static ClientConfiguration parseArgs(String[] args) throws Exception {
...
}

// Then if the application catches the exception they can call this method to print usage.
public static void printUsage() {
...
}
```

WDYT?